### DISCUSSION TITLE
core: is there still a need for `quick-pick-service` in the backend (node)?

### DISCUSSION THREAD

**Discussion by vince-fugnitto** [eclipse-theia/theia#10263] [Q&A]
I believe that in the past we needed `quick-pick-service` in the backend (https://github.com/eclipse-theia/theia/commit/1592cc5ebcd9f71a9def92340d00cdb4c544507f I believe legacy purposes for `@theia/java`):

https://github.com/eclipse-theia/theia/blob/3ccbfb0fdb3ed429f6153287bb454e6cccf9b5b3/packages/core/src/node/backend-application-module.ts#L50-L52

I'm wondering if there is still this need. I was refactoring to fix issues related to `runtime-import-check`, and came across this use-case. Is it safe to remove? Do we expect the backend to be able to call the `quick-pick-service` directly?

cc @tsmaeder @kittaakos

**Comment by tsmaeder** [comment]
I did a quick check in Che, and we're not using it from the back end. Anyway...extension writers would always have the option to bind their own "QuickPickFrontendService", right?

**Reply by vince-fugnitto** [reply]
> I did a quick check in Che, and we're not using it from the back end. Anyway...extension writers would always have the option to bind their own "QuickPickFrontendService", right?

I believe so yes.
For additional information, the issue that I'm trying to fix is how to best handle this instance of `runtime-import-check`:

https://github.com/eclipse-theia/theia/blob/3ccbfb0fdb3ed429f6153287bb454e6cccf9b5b3/packages/core/src/common/quick-pick-service.ts#L17-L18

We either move `quick-pick-service` to browser, or we make the interfaces available in `common`.
